### DISCUSSION TITLE
Using Krita Diffusion for Animation Frame Coloring – A Technique That Will Help Speed Up Production and Improve Line Layer Refinement

### DISCUSSION THREAD

**Discussion by hnanimation** [Acly/krita-ai-diffusion#1442] [Ideas]
Hello [krita diffusion Team],

First of all, I want to thank you for the excellent work you are doing on the Krita Diffusion plugin. I have noticed some points that I believe could be improved to make the coloring process more accurate and efficient. Below are my suggestions and feedback:

1. Accuracy of Using Refine Layers as Line Reference:

Currently, when I set the Refine layer for the lines to the maximum, the AI does not color the lines accurately. Instead, it re-generates them, altering the original lines and affecting the overall quality of the artwork. I would like to see improvements in how this layer is used, ensuring it acts as a reference for the lines only without re-generating them.

2. Coloring with Precision According to the Reference:

There are some issues with coloring certain parts, where the AI either fails to color them as in the reference or swaps color placements. I would appreciate an enhancement to ensure the coloring follows the reference more precisely.

3. Adjusting Generation Strength for Faster Processing:

While I appreciate the addition of an option to adjust the generation strength, coloring with high strength (e.g., 85-100%) takes a long time and is unnecessary, especially when working with flat color layers only. I would suggest providing an option to reduce the generation strength for faster and more efficient flat color application.

4. Mitigating the Fading Effect When Reducing Strength:

When the generation strength is reduced, the colors become faded, as if the color reference has become more transparent. This results in the final output being more transparent and dull compared to the original colors in the Refine layer. I would appreciate adjustments to keep the original colors more accurate.

5. Generating Colors Only Without Shadows:

I do not need to generate shadows in this process, as I will add them manually or in a separate layer later on. If possible, I would like an option to generate only the flat colors, with the shadows being generated in a separate layer without re-generating the entire image.

6. Applying the Generated Layer Manually and Avoiding Automatic Frame Transition:

I need the option to manually apply the generated layer so that the result appears directly without automatically transitioning to the next frame in the same layer. This will help me review and make adjustments before moving to the next frame.

7. Automatic Color Generation for the Entire Timeline:

I would like the ability to generate colors automatically for the selected layer across the entire timeline with one click after selecting the start and end frames. This would save a significant amount of time and effort in the coloring process.

Thank you in advance for considering this feedback, and I look forward to seeing future improvements. These adjustments will help make using Krita Diffusion more effective and precise, speeding up animation production and enhancing the quality of line layers.

Best regards,
[Hassan ,2D digital artist and animator]

**Comment by freecoderwaifu** [comment]
Color consistency in general is still the holy grail. The VAE (re)calculates colors every time, so colors will always degrade just by the nature of diffusion. The ability to define a very specific palette at the start and keep it 100% consistent would be a game changer.

Closest thing is this node, but I haven't tested it.
https://github.com/45uee/ComfyUI-Color_Transfer 

For 5) you could meanwhile use "flat colors" in your prompts and maybe a flat colors Lora, and then look for a custom workflow that does layer separation.
